### ISSUE TITLE
Failed to execute 'importScripts' on 'WorkerGlobalScope'

### ISSUE THREAD

**Issue by kishorsharma** [kishorsharma/webpack-bundle-optimizer#6]
sw.js:Uncaught DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope':

This occurs when people are using importScript in service worker. As this also became part of string however importScript() must be placed outside of a function. 

One solution is to white list only those file which you want to optimise:

```JS
// webpack.config.js
const Optimizer = require("webpack-bundle-optimizer");
module.exports = {
  entry: //...,
  output: //...,
  plugins: [
    new Optimizer({
    whiteList: ['vendor-react', 'vendor-all', 'runtime', 'main']
  })
  ]
}
```
Or blackList service worker file always:

```JS
// webpack.config.js
const Optimizer = require("webpack-bundle-optimizer");
module.exports = {
  entry: //...,
  output: //...,
  plugins: [
    new Optimizer({
    whiteList: ['sw.js']
  })
  ]
}
```

**Comment by kishorsharma** [comment]
As different project might have different naming pattern, We also allow to provide a predicate function which will be used by both whiteList and blackList checking of file.
This example assume fileName are generated with hash (which is common pattern now):
filename : main-98becfc0b54581da9521.js
```JS
// webpack.config.js
const Optimizer = require("webpack-bundle-optimizer");
module.exports = {
  entry: //...,
  output: //...,
  plugins: [
    new Optimizer({
    whiteList: ['vendor-react', 'vendor-all', 'runtime', 'main'],
    predicate: function (pName, cName) {
      const fileNameParts = cName.split('-');
      fileNameParts.pop();
      const fileName = fileNameParts.join('-');
      return pName === fileName;
    }
  })
  ]
}

```
Here as I am removing hash and extension I am passing name in whiteList array without extension. 

By default, complete filename will be matched with in whiteList array.

**Comment by kishorsharma** [comment]
Fixed in version: 1.0.2
